armory/docker/management.py

```diff
@@ -21,22 +21,33 @@ class ArmoryInstance(object):
     def __init__(
         self, image_name, runtime: str = "runc", envs: dict = None, ports: dict = None
     ):
-        self.docker_client = docker.from_env()
+        host_paths = HostPaths()
+        docker_paths = DockerPaths()
+        self.docker_client = docker.from_env(version="auto")
 
-        os.makedirs(paths.OUTPUTS, exist_ok=True)
         container_args = {
             "runtime": runtime,
             "remove": True,
             "detach": True,
             "volumes": {
-                paths.CWD: {"bind": "/workspace", "mode": "rw"},
-                paths.USER_ARMORY: {"bind": "/root/.armory", "mode": "rw"},
-                paths.USER_KERAS: {"bind": "/root/.keras", "mode": "rw"},
+                host_paths.cwd: {"bind": "/workspace", "mode": "rw"},
+                host_paths.armory_dir: {"bind": docker_paths.armory_dir, "mode": "rw"},
+                host_paths.dataset_dir: {
+                    "bind": docker_paths.dataset_dir,
+                    "mode": "rw",
+                },
+                host_paths.output_dir: {"bind": docker_paths.output_dir, "mode": "rw"},
             },
         }
         if ports is not None:
             container_args["ports"] = ports
 
+        # Windows docker does not require syncronizing file and
+        # directoriy permissions via uid and gid.
+        if os.name != "nt":
+            user_id = os.getuid()
+            container_args["user"] = f"{user_id}:{user_id}"
```

Comment:
Apparently the user id and group id associated with the user's group may not always be the same. On archimedes the group id associated with some directories is a group owned by a different user.

Reply:
Thx for the catch! Pushed a fix.